[BUGFIX] Fix for game crashing with any leader den outsider interaction

KiraDragoness opened the pull request:
## About The Pull Request

Fixed the game crashing on any interaction with outisders, but reverting the name of the actions to pre-localization.

Fixed interactions with lost cats specifically. They probably broke when the change to keep track of the previous status in the clan was introduced. The game was checking if their status was "lost" and failing, because the lost status is now determined by the boolean cat.outside. So in the check status function, I added a conditional that compares the required lost status with this boolean.

## Linked Issues
Fixes: #3379
Fixes: #3192

## Proof of Testing
![image](https://github.com/user-attachments/assets/89ca6d74-496a-4a85-8b85-aeef56ffbfb6)
![image](https://github.com/user-attachments/assets/5ff34ae6-47b4-4b90-afc8-1c3cb263c401)

## Changelog/Credits
[BUGFIX]: Fixed crash related to leader den interactions with outsiders

KiraDragoness committed:
Fix for herb_supply random crash

Apparently, this function was expecting a tuple but the entire_supply property returns a dict, causing the random crash. By adding the items method the dictionary returns a tuple.

KiraDragoness committed:
Update handle_short_events.py

KiraDragoness committed:
Fix for outsider interactions crashing the game

Fixed the name change for the outsider actions causing a crash.

Fixed interactions with lost cats by .

j-gynn commented on scripts/events_module/event_filters.py:
```diff
@@ -272,6 +272,9 @@ def _check_cat_status(cat, statuses: list) -> bool:
     if cat.status in statuses:
         return True
 
+    if 'lost' in statuses and cat.outside:
```
```suggestion
    if 'lost' in statuses and cat.status not in ["rogue", "loner", "kittypet", "former Clancat"] and cat.outside:
```

KiraDragoness replied:
My bad, I didn't think of that! Fixed now

j-gynn reviewed (changes requested):
Adjust the filter slightly so it can't pick up loners, kittypets, rogues and former Clancats, and we should be good to go.

KiraDragoness committed:
Adjusted check status function

When trying to see if the cat is lost, it will filter out any other type of outsider.
